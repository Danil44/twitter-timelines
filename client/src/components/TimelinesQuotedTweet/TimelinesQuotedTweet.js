import React from 'react';
import {
  Card,
  CardHeader,
  Avatar,
  Typography,
  CardContent,
} from '@material-ui/core';
import PropTypes from 'prop-types';
import moment from 'moment';
import useStyles from './TimelinesQuotedTweetStyle';

const TimelinesQuotedTweet = ({ user, created_at: createdAt, text }) => {
  const classes = useStyles();
  const formatedTime = moment(createdAt).format('LL');
  return (
    <Card className={classes.card}>
      <CardHeader
        avatar={<Avatar aria-label={user.name} src={user.profile_image_url} />}
        title={user.name}
        titleTypographyProps={{
          display: 'inline',
          className: classes.name,
        }}
        subheaderTypographyProps={{ display: 'inline' }}
        subheader={`@${user.screen_name} . ${formatedTime}`}
      />
      <CardContent>
        <Typography variant="body1">{text}</Typography>
      </CardContent>
    </Card>
  );
};

TimelinesQuotedTweet.propTypes = {
  user: PropTypes.shape({
    name: PropTypes.string.isRequired,
    screen_name: PropTypes.string.isRequired,
    profile_image_url: PropTypes.string.isRequired,
  }).isRequired,
  created_at: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
};

export default TimelinesQuotedTweet;
