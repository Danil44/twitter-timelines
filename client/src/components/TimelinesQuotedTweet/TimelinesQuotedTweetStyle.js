import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles({
  card: {
    maxWidth: '100%',
  },

  name: {
    marginRight: 5,
  },

  count: {
    marginRight: 15,
  },
});

export default useStyles;
