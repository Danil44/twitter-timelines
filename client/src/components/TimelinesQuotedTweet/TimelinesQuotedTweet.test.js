import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import TimelinesQuotedTweet from './TimelinesQuotedTweet';

describe('<TimelinesQuotedTweet/> component', () => {
  let wrapper;
  let props;

  beforeEach(() => {
    props = {
      user: {
        name: 'name',
        screen_name: 'screen name',
        profile_image_url: 'profile image url',
      },
      created_at: 'date',
      text: 'text',
    };

    wrapper = shallow(<TimelinesQuotedTweet {...props} />);
  });

  it('should renders without crashing given the required props', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
