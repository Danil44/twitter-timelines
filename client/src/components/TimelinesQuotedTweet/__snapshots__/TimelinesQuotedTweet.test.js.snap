// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TimelinesQuotedTweet/> component should renders without crashing given the required props 1`] = `
<WithStyles(ForwardRef(Card))
  className="makeStyles-card-1"
>
  <WithStyles(ForwardRef(CardHeader))
    avatar={
      <ForwardRef(WithStyles)
        aria-label="name"
        src="profile image url"
      />
    }
    subheader="@screen name . Invalid date"
    subheaderTypographyProps={
      Object {
        "display": "inline",
      }
    }
    title="name"
    titleTypographyProps={
      Object {
        "className": "makeStyles-name-2",
        "display": "inline",
      }
    }
  />
  <WithStyles(ForwardRef(CardContent))>
    <WithStyles(ForwardRef(Typography))
      variant="body1"
    >
      text
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(CardContent))>
</WithStyles(ForwardRef(Card))>
`;
